### PR TITLE
Make moving window dimensionality consistent in PICMI

### LINKED ISSUE #2702: PICMI: Dimensionality of `moving_window_velocity`
There seems to be a small inconsistency in how we handle the dimensionality of `moving_window_velocity` at the PICMI level.

In 1D we do
https://github.com/ECP-WarpX/WarpX/blob/eba5447a30bbbf71d729d7010ee3cf591dc3c016/Python/pywarpx/picmi.py#L504-L506
which requires to pass something like `moving_window_velocity = [0., 0., c]` as argument of `Cartesian1DGrid`.

In 2D we do
https://github.com/ECP-WarpX/WarpX/blob/eba5447a30bbbf71d729d7010ee3cf591dc3c016/Python/pywarpx/picmi.py#L559-L564
which requires to pass something like `moving_window_velocity = [0., c]` as argument of `Cartesian2DGrid`.

In 3D we do
https://github.com/ECP-WarpX/WarpX/blob/eba5447a30bbbf71d729d7010ee3cf591dc3c016/Python/pywarpx/picmi.py#L626-L634
which requires to pass something like `moving_window_velocity = [0., 0., c]` as argument of `Cartesian3DGrid`.

So, if I understand correctly, we have to pass a three-dimensional velocity in 1D and 3D and a two-dimensional velocity in 2D.

I think it would be worth cleaning this up in such a way that we either (i) match the dimensionality of the moving window velocity to the dimensionality of the geometry or (ii) always pass a three-dimensional moving window velocity regardless of the dimensionality of the geometry.

@dpgrote @ax3l @RemiLehe @prkkumar What do you think?

### PR DESCRIPTION
**Pull request by prkkumar**
This PR makes moving window velocity dimensions consistent with the code's dimensions (cf Issue #2702 ). Now, we need to use (assuming moving window direction is "z")

moving_window_velocity = [0., 0., c] in 3D
moving_window_velocity = [0., c] in 2D
moving_window_velocity = [c] in 1D

This closes #2702.

### CODE REVIEW

**Review by EZoni — APPROVED** [review]
Thank you for this PR, Prabhat! Looks good to me.

Question for @prkkumar, @ax3l and @dpgrote: 
Do we want to change also the corresponding parameter used in RZ geometry, namely `moving_window_zvelocity`?

**Comment by ax3l** [comment]
>  Do we want to change also the corresponding parameter used in RZ geometry, namely moving_window_zvelocity?

Sounds good to me, keeps them consistent :)

**Comment by dpgrote** [comment]
This looks good to me. Regarding RZ, I used the name `moving_window_zvelocity` since it is different than the other parameters which take a list of length 2 since RZ is 2D. Using `moving_window_velocity` and expecting a list of length 1 is inconsistent, and expecting an list of length 2 is unnecessary verbiage.

**Review by dpgrote — APPROVED** [review]
This looks good.

File: Python/pywarpx/picmi.py [hunk 1]
```diff
@@ -444,14 +444,14 @@ def initialize_inputs(self):
...
 
-        if self.moving_window_zvelocity is not None:
-            if np.isscalar(self.moving_window_zvelocity):
-                if self.moving_window_zvelocity !=0:
-                    pywarpx.warpx.do_moving_window = 1
-                    pywarpx.warpx.moving_window_dir = 'z'
-                    pywarpx.warpx.moving_window_v = self.moving_window_zvelocity/constants.c  # in units of the speed of light
-            else:
-                raise Exception('RZ PICMI moving_window_velocity (only available in z direction) should be a scalar')
+        if self.moving_window_velocity is not None and np.any(np.not_equal(self.moving_window_velocity, 0.)):
```
Comment (prkkumar) [inline Python/pywarpx/picmi.py:447]: In `PICMI_CylindricalGrid` class, `moving_window_zvelocity` should be changed to `moving_window_velocity`. This has to be done in a new PR in PICMI. Is there anything else needed for this? @dpgrote @EZoni

Comment (EZoni) [reply Python/pywarpx/picmi.py:447]: Thank you, Prabhat. I can look at what needs to be done on the PICMI side, in WarpX I think we might still need to update one PICMI test, in particular this line:
https://github.com/ECP-WarpX/WarpX/blob/daa51549ff6de4d20db1b388251a234bac7c2027/Examples/Physics_applications/laser_acceleration/PICMI_inputs_rz.py#L36
Things in CI will work only after the change on the PICMI side is done and this PR is rebased. I will try to address this soon.

Comment (EZoni) [reply Python/pywarpx/picmi.py:447]: @prkkumar @dpgrote This should be the PR needed on the PICMI side: https://github.com/picmi-standard/picmi/pull/54.

Comment (prkkumar) [reply Python/pywarpx/picmi.py:447]: Thanks, Edoardo! I Updated the RZ laser acceleration test. Will rebase the PR once picmi-standard/picmi#54 is merged.

**Comment by dpgrote** [comment]
I've merged the PR in picmi. I also updated the version number of pimcistandard to 0.0.19 and pushed it to PyPI. You will need to update the version number (to 0.0.19) in the two files Python/setup.py and requirements.txt. Once that is done, the CI tests should work.

**Comment by EZoni** [comment]
@prkkumar Thank you for updating the PICMI version. I just realized that there is one more PICMI test (`Python_Langmuir_rz_multimode`) where `moving_window_zvelocity` needs to be replaced with the new syntax:
https://github.com/ECP-WarpX/WarpX/blob/a3e93b5bd9587e747c6e63d0ef9cb4f298ae845c/Examples/Tests/Langmuir/PICMI_inputs_langmuir_rz_multimode_analyze.py#L94

**Review by EZoni — APPROVED** [review]

**Pull request merged**